File: 0189-rotate-array/0189-rotate-array.c
void rotate(int* nums, int numsSize, int k) {
    int n=numsSize;
  while (k>=n) {
        k-=n;  
    }
    int a[n];
    for(int i=0;i<k;i++) 
    {
        a[i]=nums[n-k+i];
    }
    for (int i=0;i<n-k;i++) 
    {
        a[i+k]=nums[i];
    }
    for (int i=0;i<n;i++)
    {
        nums[i]=a[i];
    }
}

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        List<Integer> b=new ArrayList<>();
        for(int i:nums)
            b.add(i);
        Collections.rotate(b,k);
        int c=0;
    for(int j:b)
        nums[c++]=j;
        
    }
}